recipes/frugen/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import get, rmdir, apply_conandata_patches, export_conandata_patches
+import os
+
+class FrugenConan(ConanFile):
+    name = "frugen"
+    description = "IPMI FRU Information generator / editor tool and library"
+    license = ("Apache-2.0", "GPL-2.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://codeberg.org/IPMITool/frugen"
+    topics = ("hardware", "ipmi", "fru")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_json": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_json": True
+    }
+    implements = ["auto_shared_fpic"]
+
+    def validate(self):
+        if self.settings.os in ["Windows", "Macos"]:
+            raise ConanInvalidConfiguration("Unsupported os. This package currently does not support Macos and Windows")
```

Comment:
I think windows _is_ supported (as per the documentation)  just not with MSVC + Windows SDK -  please try to avoid invalidating platforms just to get CI to pass - it gives us an opportunity to look at what's wrong and advise

Reply:
I bet no one tested latest release under Windows, even with MinGW =). But you right, it should work. What can I refer as an example on how to do this better?

Reply:
Thanks @alatarum !! What I mean is that - invalidating is "probably" fine - but as reviewers we need to see why it is being invalidated. Sometimes it is 100% true that indeed it is an "unsupported OS" - this can be evidenced by going to the documentation, or similar - but other times, it fails for other reasons (could be something related to our CI infrastructure, or something else) - and more often than not we end up with a PR a few days/weeks/months down the line removing a "too strict" invalidation :/ 

What we need to see is the supporting evidence behind the "unsupported os" - so that we can advise to tailor what needs to be invalidated, for example
- "doesn't work on windows with msvc because it requires sys/time.h"
- "doesnt work on macOS because it includes the wrong header and there is little evidence that it was ever used or tested on macOS"

That leaves a "paper trail" of sorts, for us to see, and for everyone else in the future when they need to reason why something was done :) 

And hopefully to avoid misunderstanding - we don't expect contributors to test or try to fix platforms where CI is failing  - I appreciate your efforts on this front on macOS! Sometimes patches are trivial - but a lot of times they aren't, and they lead us to believe that certain variants of libraries would _only_ exist in Conan, and that's not our goal - quite the opposite in fact, as that would also mean that we would we providing something that was never fully tested _outside_ of the build process of Conan Center, which is quite a risk!